Replace leftover joomla.org (and subdomains) HTTP links for HTTPS links (where possible)

andrepereiradasilva opened the pull request:
#### Description

An year after https://github.com/joomla/joomla-cms/pull/5478, this PR is for normalize the joomla.org URI in Joomla code to use HTTPS. 

In this PR all joomla.org domains (and subdomains) in joomla code were checked, and then checked if those serve in HTTP, HTTP and HTTPS or exclusively HTTPS (HTTP redirect to HTTPS).

For the ones that the change seems possible without issues (certificate errors, protocols errors, redirects to HTTP, mixed content, etc) the change is done in this PR.

Please note this is not a blind search -> replace.
##### List of joomla.org domains (found in code and completly https with this PR)

| Domain | Protocols | Observations | All URI in HTTPS? |
| :-- | :-: | :-- | :-: |
| joomla.org | HTTP and HTTPS | Forces HTTPS. Redirect to www.joomla.org | :white_check_mark: (after this PR) |
| www.joomla.org | HTTP and HTTPS | Forces HTTPS | :white_check_mark: (after this PR) |
| docs.joomla.org | HTTP and HTTPS | Forces HTTPS | :white_check_mark: (after this PR) |
| developer.joomla.org | HTTP and HTTPS | Forces HTTPS | :white_check_mark: (after this PR) |
| volunteers.joomla.org | HTTP and HTTPS | Forces HTTPS | :white_check_mark: (after this PR) |
| events.joomla.org | HTTP and HTTPS | Forces HTTPS | :white_check_mark: (after this PR) |
| issues.joomla.org | HTTP and HTTPS | Forces HTTPS | :white_check_mark: (after this PR) |
| api.joomla.org | HTTP and HTTPS | Forces HTTPS | :white_check_mark: |
| demo.joomla.org | HTTP and HTTPS | Forces HTTPS | :white_check_mark: |
| contribute.joomla.org | HTTP and HTTPS | Forces HTTPS. Redirect to opensourcematters.org | :white_check_mark: (after this PR) |
| help.joomla.org | HTTP and HTTPS | Forces HTTPS in both (/ and /proxy/) | :white_check_mark: (after this PR) |
| cdn.joomla.org | HTTP and HTTPS | - | :white_check_mark: (after this PR) |
##### List of joomla.org domains (found in code but will have to wait for a future PR)

| Domain | Protocols | Observations | All URI in HTTPS? |
| :-- | :-: | :-- | :-: |
| update.joomla.org | HTTP and HTTPS | - | PR related to this: https://github.com/joomla/joomla-cms/pull/8645 |
| shop.joomla.org | HTTP and HTTPS | HTTPS has Mixed content. | PR related to this: https://github.com/joomla/joomla-cms/pull/8653 |
| community.joomla.org | HTTP and HTTPS | HTTPS has Mixed content. | Site not ready yet. |
| resources.joomla.org | HTTP and HTTPS | HTTPS has Mixed content. | Site not ready yet. |
| extensions.joomla.org | HTTP and HTTPS | HTTPS redirect to HTTP | Site not ready yet. |
| forum.joomla.org | HTTP and HTTPS | HTTPS has protocol error. | Site not ready yet. |
| opensourcematters.org | HTTP | - | Site not ready yet. |
| feeds.joomla.org | HTTP | Feedburner | Not ready yet. |
| build.joomla.org | HTTP and HTTPS | Jenkings. Certificate error | Not ready yet. |
| joom.la | HTTP and HTTPS | Certificate error | Not ready yet. |
##### Other domain in code

There are other domains used only for Unit tests to simulate errors but doesn't really exist or are only used in unit tests, so those were not changed:
- base.joomla.org
- test.joomla.org
- xommunity.joomla.org
#### How to test

Check the "Files changed" in this PR and verify the changes.
#### More info

For checking the domains that exist in Joomla code it was used several `egrep` linux commands:
- `egrep -R 'joomla\.org' /path/to/joomla/`
- `egrep -R 'http:(//|\\/\\/|\\\\/\\\\/)subdomain\.joomla\.org' /path/to/joomla/`
- `egrep -R 'https:(//|\\/\\/|\\\\/\\\\/)subdomain\.joomla\.org' /path/to/joomla/`
- and other combinations

See https://github.com/joomla/joomla-websites/issues/20 for a list of Joomla domains and the HTTP to HTTPS site conversion process.

Suggestions or corrections are welcome.

andrepereiradasilva committed:
Merge remote-tracking branch 'refs/remotes/joomla/staging' into staging

Commit:
Merge remote-tracking branch 'refs/remotes/joomla/staging' into staging

Commit:
Merge remote-tracking branch 'refs/remotes/joomla/staging' into staging

Commit:
Merge branch 'staging' of https://github.com/andrepereiradasilva/joomla-cms into staging

andrepereiradasilva committed:
Merge remote-tracking branch 'refs/remotes/joomla/staging' into more-https-links

andrepereiradasilva committed:
From http to https

wojsmol commented:
@andrepereiradasilva  regarding cdn see first post of https://github.com/joomla/joomla-websites/issues/20

andrepereiradasilva committed:
some missing in json format

mbabker commented:
For help.joomla.org the help screen proxy (/proxy on that subdomain)
already defaults to HTTPS and the rest of the site is effectively offline
but the htaccess should be forcing HTTPS on the landing page last I
remember.  So that's fine to change any leftover links.  Though those links
should probably change to something suitable in the docs wiki.

On Saturday, December 26, 2015, andrepereiradasilva <
notifications@github.com> wrote:

> Description
>
> An year after #5478 <https://github.com/joomla/joomla-cms/pull/5478>,
> this PR is for normalize the joomla.org URI in Joomla code to use https.
>
> In this PR all joomla.org domains (and subdomains) in joomla code were
> checked, and then checked if those serve in HTTP, HTTP and HTTPD or
> exclusively HTTPS (HTTP redirect to HTTPS).
>
> For the ones that the change seems possible without issues (certificate
> errors, protocols errors, redirects to HTTP, mixed content, etc) the change
> is done in this PR.
>
> Please note this is not a blind search -> replace.
> List of joomla.org domains Domain Protocols Observations All URI in HTTPS?
> joomla.org HTTP and HTTPS Both redirect to www.joomla.org [image:
> :white_check_mark:] (after this PR) www.joomla.org HTTP and HTTPS HTTP
> redirect to HTTPS [image: :white_check_mark:] (after this PR)
> docs.joomla.org HTTP and HTTPS HTTP redirect to HTTPS [image:
> :white_check_mark:] (after this PR) developer.joomla.org HTTP and HTTPS HTTP
> redirect to HTTPS [image: :white_check_mark:] (after this PR)
> volunteers.joomla.org HTTP and HTTPS HTTP redirect to HTTPS [image:
> :white_check_mark:] (after this PR) issues.joomla.org HTTP and HTTPS HTTP
> redirect to HTTPS [image: :white_check_mark:] (after this PR)
> api.joomla.org HTTP and HTTPS HTTP redirect to HTTPS [image:
> :white_check_mark:] demo.joomla.org HTTP and HTTPS HTTP redirect to HTTPS [image:
> :white_check_mark:] contribute.joomla.org HTTP and HTTPS Both redirect to
> opensourcematters.org [image: :white_check_mark:] (after this PR)
> update.joomla.org HTTP and HTTPS - PR related to this: #8645
> <https://github.com/joomla/joomla-cms/pull/8645> help.joomla.org HTTP and
> HTTPS - Code not ready yet? Same as above? shop.joomla.org HTTP and HTTPS HTTPS
> has Mixed content. PR related to this: #8653
> <https://github.com/joomla/joomla-cms/pull/8653> community.joomla.org HTTP
> and HTTPS HTTPS has Mixed content. Site not ready yet.
> resources.joomla.org HTTP and HTTPS HTTPS has Mixed content. Site not
> ready yet. extensions.joomla.org HTTP and HTTPS HTTPS redirect to HTTP Site
> not ready yet. forum.joomla.org HTTP and HTTPS HTTPS has protocol error. Site
> not ready yet. feeds.joomla.org HTTP Feedburner Not ready yet.
> build.joomla.org HTTP and HTTPS Jenkings. Certificate error Not ready yet. Other
> domain in code
>
> There are other domains used only for Unit tests to simulate errors but
> doesn't really exist or are only used in unit tests, so those were not
> changed:
>
>    - cdn.joomla.org
>    - base.joomla.org
>    - test.joomla.org
>    - xommunity.joomla.org
>
> How to test
>
> Check the "Files changed" in this PR and verify the changes.
> More info
>
> For checking the domains that exist in Joomla code it was used several
> egrep linux commands:
>
>    - egrep -R 'joomla\.org' /path/to/joomla/
>    - egrep -R 'http:(//|\/\/)subdomain\.joomla\.org' /path/to/joomla/
>    - egrep -R 'https:(//|\/\/)subdomain\.joomla\.org' /path/to/joomla/
>    - and other combinations
>
> Suggestions or corrections are welcome.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/joomla/joomla-cms/pull/8791
> Commit Summary
>
>    - Merge remote-tracking branch 'refs/remotes/joomla/staging' into
>    staging
>    - Merge remote-tracking branch 'refs/remotes/joomla/staging' into
>    staging
>    - Merge remote-tracking branch 'refs/remotes/joomla/staging' into
>    staging
>    - Merge branch 'staging' of
>    https://github.com/andrepereiradasilva/joomla-cms into staging
>    - Merge remote-tracking branch 'refs/remotes/joomla/staging' into
>    more-https-links
>    - From http to https
>
> File Changes
>
>    - *M* README.md
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-0> (2)
>    - *M* README.txt
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-1> (2)
>    - *M* installation/sql/mysql/sample_learn.sql
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-2> (6)
>    - *M* installation/sql/mysql/sample_testing.sql
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-3> (10)
>    - *M* installation/sql/postgresql/sample_data.sql
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-4> (2)
>    - *M* installation/sql/postgresql/sample_learn.sql
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-5> (6)
>    - *M* installation/sql/postgresql/sample_testing.sql
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-6> (10)
>    - *M* installation/sql/sqlazure/sample_data.sql
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-7> (2)
>    - *M* installation/sql/sqlazure/sample_learn.sql
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-8> (6)
>    - *M* installation/sql/sqlazure/sample_testing.sql
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-9> (18)
>    - *M* libraries/cms/version/version.php
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-10> (2)
>    - *M* libraries/joomla/cache/storage/file.php
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-11> (2)
>    - *M* libraries/joomla/document/feed/renderer/atom.php
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-12> (2)
>    - *M* tests/system/webdriver/Pages/System/AdminPage.php
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-13> (4)
>    - *M* tests/system/webdriver/tests/report/index.php
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-14> (2)
>    - *M* tests/unit/stubs/database/jos_banners.csv
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-15> (2)
>    - *M* tests/unit/stubs/database/jos_content.csv
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-16> (2)
>    - *M* tests/unit/stubs/database/jos_finder_links.csv
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-17> (2)
>    - *M* tests/unit/stubs/database/jos_menu.csv
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-18> (2)
>    - *M* tests/unit/stubs/database/jos_ucm_content.csv
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-19> (2)
>    - *M* tests/unit/suites/libraries/joomla/document/JDocumentTest.php
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-20> (4)
>    - *M*
>    tests/unit/suites/libraries/joomla/document/feed/renderer/JDocumentRendererAtomTest.php
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-21> (2)
>    - *M*
>    tests/unit/suites/libraries/joomla/document/html/JDocumentHTMLTest.php
>    <https://github.com/joomla/joomla-cms/pull/8791/files#diff-22> (2)
>
> Patch Links:
>
>    - https://github.com/joomla/joomla-cms/pull/8791.patch
>    - https://github.com/joomla/joomla-cms/pull/8791.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/8791>.
>

andrepereiradasilva commented:
@mbabker 
The ones that are left from `help.joomla.org` domain in joomla code are:

- https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/help/help.php#L187
- https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_config/controller/application/refreshhelp.php#L41
- https://github.com/joomla/joomla-cms/blob/staging/tests/unit/suites/libraries/cms/help/JHelpTest.php#L118

I think those can also be changed to HTTPS without issues. Do you agree?

mbabker commented:
Yes those can change.

On Saturday, December 26, 2015, andrepereiradasilva <
notifications@github.com> wrote:

> @mbabker <https://github.com/mbabker>
> The ones that are left from help.joomla.org domain in joomla code are:
>
>    -
>    https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/help/help.php#L187
>    -
>    https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_config/controller/application/refreshhelp.php#L41
>    -
>    https://github.com/joomla/joomla-cms/blob/staging/tests/unit/suites/libraries/cms/help/JHelpTest.php#L118
>
> I think those can also be changed to HTTPS without issues. Do you agree?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/8791#issuecomment-167356623>.
>

andrepereiradasilva committed:
Update help.joomla.org to HTTPS too

mbabker commented on tests/system/webdriver/Pages/System/AdminPage.php:
```diff
@@ -140,8 +140,8 @@
 			'Community Portal'		=> 'http://community.joomla.org/',
-			'Security Center'		=> 'http://developer.joomla.org/security.html',
-			'Developer Resources'	=> 'http://developer.joomla.org/',
+			'Security Center'		=> 'https://developer.joomla.org/security.html',
```
This should actually be `Security Centre` and linking to `https://developer.joomla.org/security-centre.html`.  Not 100% related to this PR but this particular test has been failing for some time if it's comparing links and text.

mbabker commented:
As for the test case involving `cdn.joomla.org` it's validating a behavior in that class' API and not reliant on a "real" page.  You can try changing it to HTTPS but I think the protocol is dependent on the injected config values in that test case IIRC so it might end up failing without additional changes.

andrepereiradasilva committed:
updated security center URI

andrepereiradasilva committed:
cdn.joomla.org to HTTPS

andrepereiradasilva commented:
@mbabker thanks for your comments

Made the two changes and travis accepted them without problems.

We also have the `joom.la` domain (only in https://github.com/joomla/joomla-cms/blob/staging/tests/unit/suites/libraries/joomla/application/JApplicationWebTest.php) that i think is some kind of URI shortener service, right?

By the way, the certificate is incorrect in that domain.

andrepereiradasilva committed:
update string size

infograf768 commented:
Looking at this patch and specially the tests/unit/stubs/database/...
I found 27 occurences of these unknown characters
>�*�

mbabker commented:
It's some kind of quirk unrelated to this patch (it's actually a `\u0` character according to Chrome).  This screenshot is me putting an `echo serialize($item);die;` line in before rows are written to the table where those characters are being found.

<img width="1440" alt="screen shot 2015-12-27 at 11 15 35 am" src="https://cloud.githubusercontent.com/assets/368545/12011008/6a57bef6-ac8b-11e5-9d47-cff5916d4cae.png">

mbabker commented:
It's PHP's behavior actually... http://php.net/manual/en/function.serialize.php

> Object's private members have the class name prepended to the member name; protected members have a '*' prepended to the member name. These prepended values have null bytes on either side.

infograf768 commented:
Thanks Michael. Looks like it is the same as
http://apps.timwhitlock.info/unicode/inspect?s=%EF%BF%BD#note1

andrepereiradasilva commented:
Ok. So all fine.

andrepereiradasilva committed:
Replace events site URI

Replace: http://community.joomla.org/events.html
for: https://events.joomla.org/

andrepereiradasilva commented:
closed as comments above, new PR will be made.
